src/library.js

```diff
@@ -2955,7 +2955,7 @@ addToLibrary({
     }
 #endif
 #if ASSERTIONS
-    if (!ASM_CONSTS.hasOwnProperty(code)) abort(`No EM_ASM constant found at address ${code}`);
+    assert(ASM_CONSTS.hasOwnProperty(code), `No EM_ASM constant found at address ${code}.  The loaded WebAssembly file is likely out of sync with the generated JavaScript.`);
```

Comment:
I wonder if this message will carry the point across? It does print what the most common cause is (out of sync), but it does not hint to the user why it could have gotten out of sync. A developer might read this print and ponder "well why did Emscripten produce an out of sync code output in the first place?", akin to an "was this an internal Emscripten failure"?

Maybe a direct message like

```
The loaded WebAssembly file is likely out of sync with the generated JavaScript. This could be caused for example by mismatched caching directives on .wasm and .js files provided by your web server. Try shift-reloading the page to clear the browser cache.
```

might help further?

Reply:
Agreed, we could give even more details.  

The amount of detail I added here is, I think, enough that if anyone reports it on the bug tracker than we (the emscripten developers) will immediately know what the problem is and be able to help them.  I feel like I wasted some time in #20329 because I didn't know the probable cause myself.   Obviously if we can avoid the bugs ever being reporting that would be even better.  If you (or anybody else) want to add more info here feel free.